activesupport/lib/active_support/messages/codec.rb

```diff
@@ -28,7 +28,7 @@ def encode(data, url_safe: @url_safe)
 
         def decode(encoded, url_safe: @url_safe)
           url_safe ? ::Base64.urlsafe_decode64(encoded) : ::Base64.strict_decode64(encoded)
-        rescue ArgumentError => error
+        rescue StandardError => error
```

Comment:
This copies the same as https://github.com/simi/rails/blob/a27a038dc323b46f53c0fd10e52a3414b6d42c6b/activesupport/lib/active_support/messages/codec.rb#L31. I'm happy to propose to Base64 gem, but I doubt it will be merged as I explained in original description.

Reply:
Just 2 cents: the trouble here is differentiating between 1) ArgumentError, you called me with an array/nil but I expected a string, aka and easily testable wrong argument that the dev should catch and 2) ArgumentError, you called me with a string that, while correct in any way you can outwardly test, it fails to evaluate.

Preventing exception 1 is easy, just check type. Preventing 2 requires reimplementing base64decode as a guard. This says to me that base64decode MUST implement a different error for 1 than 2.

1 is a programmer error, 2 requires base64decode to tell you its wrong.